tests/stats-tests.dx

```diff
@@ -0,0 +1,182 @@
+-- stats-tests.dx
+
+import stats
+
+-- mainly cross-checking against results from R
+
+-- bernoulli
+
+log_pdf (Bernoulli 0.6) True ~~ -0.5108256
+> True
+
+log_cdf (Bernoulli 0.6) False ~~ log 0.4
+> True
+
+log_sf (Bernoulli 0.6) False ~~ log 0.6
+> True
+
+quantile (Bernoulli 0.6) 0.39 == False
+> True
+
+quantile (Bernoulli 0.6) 0.41 == True
+> True
+
+rand_vec 5 (draw $ Bernoulli 0.6) (new_key 0) :: Fin 5=>Bool
+> [False, True, True, True, True]
+
+-- binomial
+
+log_pdf (Binomial 10 0.4) 8 ~~ -4.545315
+> True
+
+log_cdf (Binomial 10 0.4) 7 ~~ -0.01237076
+> True
+
+log_sf (Binomial 10 0.4) 7 ~~ -4.398599
+> True
+
+quantile (Binomial 10 0.4) 0.5 == 4
+> True
+
+rand_vec 5 (draw $ Binomial 10 0.4) (new_key 0) :: Fin 5=>Nat
+> [1, 0, 7, 6, 7]
+
+-- exponential
+
+log_pdf (Exponential 2.0) 3.0 ~~ -5.306853
```

Comment:
It might be worth having some tests for inputs outside of the domain. It's not too difficult to accidentally get something slightly below zero due to floating point inaccuracies.